classifier.go

```diff
@@ -75,31 +75,50 @@ type License struct {
 	// Threshold is the lowest confidence percentage acceptable for the
 	// classifier.
 	Threshold float64
+
+	// db is the path to the license database
+	db string
+}
+
+// OptionFunc set options on a License struct.
+type OptionFunc func(l *License) error
+
+// Database is an OptionFunc to specify the location of the license database.
+func Database(db string) OptionFunc {
+	// should we check for db existence here?
+	return func(l *License) error {
+		l.db = db
+		return nil
+	}
 }
 
 // New creates a license classifier and pre-loads it with known open source licenses.
-func New(threshold float64) (*License, error) {
+func New(threshold float64, options ...OptionFunc) (*License, error) {
 	classifier := &License{
 		c:         stringclassifier.New(threshold, Normalizers...),
 		Threshold: threshold,
+		db:        LicenseArchive,
 	}
-	if err := classifier.registerLicenses(LicenseArchive); err != nil {
-		return nil, fmt.Errorf("cannot register licenses: %v", err)
+
+	for _, o := range options {
+		err := o(classifier)
+		if err != nil {
+			return nil, fmt.Errorf("error setting option %v: %v", o, err)
+		}
+	}
+
+	if err := classifier.registerLicenses(classifier.db); err != nil {
```

Comment:
Why pass in the db as a parameter when it's part of the object?

Reply:
I was trying to minimize interface changes, but this one is private.